lib/nanoc/base/checksummer.rb

```diff
@@ -46,14 +48,14 @@ def calc(obj, digest_class = CompactDigest)
 
       private
 
-      def update(obj, digest, visited = Set.new)
+      def update(obj, digest, visited = Hamster::Set.new)
         digest.update(obj.class.to_s)
 
         if visited.include?(obj)
           digest.update('<recur>')
         else
           digest.update('<')
-          behavior_for(obj).update(obj, digest) { |o| update(o, digest, visited + [obj]) }
+          behavior_for(obj).update(obj, digest) { |o| update(o, digest, visited.add(obj)) }
```

Comment:
I could’ve left it that way, but `#add` is a nicer API, and it’s likely faster as well.